packages/frontend-2/components/settings/workspaces/Billing.vue

```diff
@@ -80,32 +192,60 @@ graphql(`
   }
 `)
 
+type SeatPrices = {
+  [key in WorkspacePlans]: {
+    [BillingInterval.Monthly]: number
+    [BillingInterval.Yearly]: number
+  }
+}
+
 const props = defineProps<{
   workspaceId: string
 }>()
 
 const isBillingIntegrationEnabled = useIsBillingIntegrationEnabled()
 const isYearlyPlan = ref(false)
+// TODO: get these from the backend when available
+const seatPrices = ref<SeatPrices>({
+  [WorkspacePlans.Team]: { monthly: 12, yearly: 12 },
+  [WorkspacePlans.Pro]: { monthly: 40, yearly: 40 },
+  [WorkspacePlans.Business]: { monthly: 79, yearly: 79 },
+  [WorkspacePlans.Academia]: { monthly: 0, yearly: 0 },
+  [WorkspacePlans.Unlimited]: { monthly: 0, yearly: 0 }
+})
 
-const checkoutUrl = (plan: string) =>
-  `/api/v1/billing/workspaces/${
-    props.workspaceId
-  }/checkout-session/${plan}/${billingCycle()}`
-const billingCycle = () => (isYearlyPlan.value ? 'yearly' : 'monthly')
-const teamCheckout = () => {
-  window.location.href = checkoutUrl('team')
-}
-const proCheckout = () => {
-  window.location.href = checkoutUrl('pro')
-}
-const businessCheckout = () => {
-  window.location.href = checkoutUrl('business')
-}
-
-const { result } = useQuery(settingsWorkspaceBillingQuery, () => ({
+const { client: apollo } = useApolloClient()
+const { result: workspaceResult } = useQuery(settingsWorkspaceBillingQuery, () => ({
   workspaceId: props.workspaceId
 }))
+const { result: pricingPlansResult } = useQuery(settingsWorkspacePricingPlansQuery)
 
-const billing = computed(() => result.value?.workspace.billing)
+const billing = computed(() => workspaceResult.value?.workspace.billing)
 const discount = computed(() => billing.value?.cost?.discount)
+const currentPlan = computed(() => workspaceResult.value?.workspace.plan)
+const subscription = computed(() => workspaceResult.value?.workspace.subscription)
+const pricingPlans = computed(
+  // Todo: fix the return type of 'workspacePricingPlans', now it's {} but does actually hold values
+  () =>
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
```

Comment:
@gjedlicska I need to a bunch of eslint-disables here, because the type of workspacePricingPlans  is `{}`, not the structure of the JSON (so `workspacePlanInformation` wont exist on `{}`), can you advice me on how to fix the type of `workspacePricingPlans` to match the expected format?

Reply:
Will be solved with the solution provided by @fabis94